resources/overlay/usr/local/bin/fast_page_fault_helper.c

```diff
@@ -0,0 +1,44 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Helper program for triggering fast page faults after UFFD snapshot restore.
+// Allocates a 128M memory area using mmap, touches every page in it using memset and then
+// calls `sigwait` to wait for a SIGUSR1 signal. Upon receiving this signal,
+// set the entire memory area to 1, to trigger fast page fault.
+// The idea is that an integration test takes a snapshot while the process is
+// waiting for the SIGUSR1 signal, and then sends the signal after restoring.
+// This way, the `memset` will trigger a fast page fault for every page in
+// the memory region.
+
+#include <stdio.h>    // perror
+#include <signal.h>   // sigwait and friends
+#include <string.h>   // memset
+#include <sys/mman.h> // mmap
+
+#define MEM_SIZE_MIB (128 * 1024 * 1024)
+
+int main(int argc, char *const argv[]) {
+    sigset_t set;
+    int signal;
+
+    sigemptyset(&set);
+    if(sigaddset(&set, SIGUSR1) == -1) {
```

Comment:
nit: can add spaces after `if`

Reply:
that is the most common indentation I've been seeing in C

Reply:
Thanks for explaining. I don't think we should drive formatting through code reviews, so if we think there's a need to have our C-files adhere to a specific formatting, we should add a test for it (and provide an automated way to apply the formatting).

Reply:
sorry to chip in. It's really a nit from my PoV so feel free to disregard what I'll say in this comment.

I agree that ideally we should have the equivalent to `cargo fmt` for C if we want to apply common style across our C files. However, setting the equivalent of `cargo fmt` for C is not as straight forward as the in the Rust case (the tooling is as standardized). That said `if(...)` does look very weird to C programmers :P